jha/src/helpers/validators.js

```diff
@@ -16,3 +21,29 @@ export const nonBCValidator = (value) => {
   return value !== 'BC'
       && value !== 'British Columbia';
 };
+
+export const dateDataRequiredValidator = (dateData) => {
+  return () => {
+    if (!dateData || !dateData.year || typeof dateData.month !== 'number' || !dateData.day) {
+      return false;
+    }
+    return true;
+  };
+};
+
+export const dateDataValidator = (dateData) => {
+  return () => {
+    if (!dateData || !dateData.year || typeof dateData.month !== 'number' || !dateData.day) {
+      return true;
+    }
+    const year = dateData.year;
+    const month = dateData.month;
+    const day = dateData.day;
+    if (!(year && typeof month === 'number' && day)
+      && (year || typeof month === 'number' || day)) {
+      return false;
+    }
+    const isoDateString = getISODateString(year, month + 1, day);
+    return isValidISODateString(isoDateString);
```

Comment:
isValidISODateString will always be true when testing the result of getISODateString unless year given to it has more than 4 digits, the day has more than 2 digits, or the month has more than two digits.  Being that the data being handed to it is originating in a DateInput Component, it is not possible for this to be the case.  Therefore dateDataValidator can only ever return true(see previous comment).

Reply:
`getISODateString` could theoretically return "YYYY-MM-32" if it's supplied those values. That's why the `isValidISODateString` is needed.